executor/runtime/docker/docker.go

```diff
@@ -350,6 +351,25 @@ func maybeAddOptimisticDad(sysctl map[string]string) {
 	}
 }
 
+func (r *DockerRuntime) computeDNSServers() []string {
+	switch r.c.EffectiveNetworkMode() {
+	case titus.NetworkConfiguration_Ipv6AndIpv4.String():
+		// True dual stack means we should provide both
+		return []string{"fd00:ec2::253", "169.254.169.253"}
+	case titus.NetworkConfiguration_Ipv6AndIpv4Fallback.String():
+		// IPv6 with fallback means we only want v6 resolvers, which reduces
+		// the burden on TSA for ipv4 udp traffic
+		return []string{"fd00:ec2::253"}
+	case titus.NetworkConfiguration_Ipv6Only.String():
+		// True ipv6 only means we really can only have a v6
+		// resolver
+		return []string{"fd00:ec2::253"}
+	default:
+		// Any other situation means we can return the classic v4 resolver
```

Comment:
I'd // TODO this as maybe defaulting to V6 later, and treating V4 only as a specific case